src/main/java/com/swrve/ratelimitedlogger/LogWithPatternAndLevel.java

```diff
@@ -163,10 +161,11 @@ private long elapsedMsecs() {
      * activity took place, this is useful enough.
      */
     private void incrementStats() {
-        if (!stats.isPresent()) {
-            return;
-        }
-        stats.get().increment(level.getLevelName() + RATE_LIMITED_COUNT_SUFFIX);
+        getStats().ifPresent(s -> s.increment(level.getLevelName() + RATE_LIMITED_COUNT_SUFFIX));
+    }
+
+    private Optional<CounterMetric> getStats() {
+        return Optional.ofNullable(stats);
```

Comment:
this will create GC churn by creating a transient Optional object.  probably better to keep it as a potentially null object and keep an ```if (stats != null) { }``` check in the incrementStats() method instead